update.go

```diff
@@ -21,92 +21,92 @@ import (
 	"strconv"
 	"strings"
 
+	"gopkg.in/src-d/go-git.v4/plumbing/transport"
+
 	git "gopkg.in/src-d/go-git.v4"
 	"gopkg.in/src-d/go-git.v4/config"
 	"gopkg.in/src-d/go-git.v4/plumbing/transport/http"
 	"gopkg.in/src-d/go-git.v4/storage/memory"
 )
 
-func update(upstream string, branch string, username string, password string) error {
-
-	// Create a new remote
-	r1 := newRemote(upstream)
-
-	// Fetch using the new remote
-	err := r1.Fetch(&git.FetchOptions{
-		RemoteName: "origin",
-	})
+func update(upstream string, branch string, username string, token string) error {
 
+	// Validate URL
+	err := validateUpstream(upstream, username, token)
 	if err != nil {
-		fmt.Println(err)
-		log.Fatalf("%s is not a valid URL\n", upstream)
+		log.Println(err)
 	}
 
 	path := extractPath(upstream)
 
-	// We instance a new repository targeting the given path (the .git folder)
-	r2, err := git.PlainInit(path, false)
-	if err != nil {
-		log.Println(err)
+	// Initialize non bare repo
+	r, err := git.PlainInit(path, false)
+	if err != git.ErrRepositoryAlreadyExists && err != nil {
+		log.Fatal(err)
 	}
 
-	fileSystemPath := path
-	r2, err = git.PlainOpen(fileSystemPath)
+	// Open repo, if initialized
+	r, err = git.PlainOpen(path)
```

Comment:
We could optimize this and not call it, when r is already a working repo (when init was run), but it would in my opinion make the code more complex and doesn't really make a huge difference. Maybe later with huge repositories, but overall it feels wrong to optimize. Just wanted to add as a bit of context.

So we have these different possible flows:
* no repo -> init -> open -> working with the repo
* repo -> init fails -> open -> working with the repo

The optimization would use:
* no repo -> init -> no open necessary -> working with the repo
* repo -> init fails -> open -> working with the repo